python_modules/dagstermill/dagstermill/__init__.py

```diff
@@ -156,33 +179,54 @@ def populate_context(
-        self.solid_def = self.pipeline_def.solid_def_named(solid_def_name)
-
+        self.solid_def_name = solid_def_name
         self.marshal_dir = marshal_dir
-        loggers = None
-        if output_log_path != 0:
-            event_logger = construct_json_event_logger(output_log_path)
-            loggers = [event_logger]
-        # do not include event_callback in ExecutionMetadata,
-        # since that'll be taken care of by side-channel established by event_logger
-        run_config = RunConfig(run_id, loggers=loggers)
-        # See block comment above referencing this issue
-        # See https://github.com/dagster-io/dagster/issues/796
-        with yield_pipeline_execution_context(
-            self.pipeline_def, environment_dict, run_config
-        ) as pipeline_context:
-            self.context = DagstermillInNotebookExecutionContext(pipeline_context)
+
+        if self.repository_def is None:
+            self.pipeline_def = PipelineDefinition([], name='Dummy Pipeline (No Repo Registration)')
+            self.input_name_type_dict = dict_to_enum(input_name_type_dict)
+            self.output_name_type_dict = dict_to_enum(output_name_type_dict)
+            for _, runtime_type_enum in self.input_name_type_dict.items():
+                if runtime_type_enum == SerializableRuntimeType.NONE:
+                    raise DagstermillError(
+                        'If Dagstermill solids have inputs that require serialization strategies '
+                        'that are not pickling, then you must register a repository within '
+                        'notebook by calling dm.register_repository(repository_def)'
+                    )
+            for _, runtime_type_enum in self.output_name_type_dict.items():
+                if runtime_type_enum == SerializableRuntimeType.NONE:
+                    raise DagstermillError(
+                        'If Dagstermill solids have outputs that require serialization strategies '
+                        'that are not pickling, then you must register a repository within notebook '
+                        'by calling dm.register_repository(repository_def).'
+                    )
+            with yield_pipeline_execution_context(
+                self.pipeline_def, {}, RunConfig(run_id=run_id)
+            ) as pipeline_context:
+                self.context = DagstermillInNotebookExecutionContext(pipeline_context)
+        else:
+            self.pipeline_def = self.repository_def.get_pipeline(pipeline_def_name)
+            check.invariant(self.pipeline_def.has_solid_def(solid_def_name))
+            self.solid_def = self.pipeline_def.solid_def_named(solid_def_name)
+
+            loggers = None
+            if output_log_path != 0:
+                event_logger = construct_json_event_logger(output_log_path)
+                loggers = [event_logger]
+            # do not include event_callback in ExecutionMetadata,
```

Comment:
do we want an explicit null/noop event callback? (`lambda *args, **kwargs: pass` or equiv?)